domain/application/state/application.go

```diff
@@ -70,32 +71,66 @@ func (st *ApplicationState) CreateApplication(ctx context.Context, name string,
 		CharmID:       charmID.String(),
 		LifeID:        life.Alive,
 	}
-	createApplication := `
-INSERT INTO application (*) VALUES ($applicationDetails.*)
-`
+	createApplication := `INSERT INTO application (*) VALUES ($applicationDetails.*)`
 	createApplicationStmt, err := st.Prepare(createApplication, appDetails)
 	if err != nil {
 		return "", errors.Trace(err)
 	}
 
+	scaleInfo := applicationScale{
+		ApplicationID: appID.String(),
+		Scale:         len(units),
+	}
+	createScale := `INSERT INTO application_scale (*) VALUES ($applicationScale.*)`
+	createScaleStmt, err := st.Prepare(createScale, scaleInfo)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
 	platformInfo := applicationPlatform{
 		ApplicationID:  appID.String(),
-		OSTypeID:       app.Platform.OSTypeID,
+		OSTypeID:       int(app.Platform.OSType),
 		Channel:        app.Platform.Channel,
-		ArchitectureID: app.Platform.ArchitectureID,
+		ArchitectureID: int(app.Platform.Architecture),
 	}
 	createPlatform := `INSERT INTO application_platform (*) VALUES ($applicationPlatform.*)`
 	createPlatformStmt, err := st.Prepare(createPlatform, platformInfo)
 	if err != nil {
 		return "", errors.Trace(err)
 	}
 
-	scaleInfo := applicationScale{
-		ApplicationID: appID.String(),
-		Scale:         len(units),
+	var (
```

Comment:
These additions are starting to make this method quite long. I can see that it's primarily a bunch of prepares and a bunch of queries.